File: typescript/6kyu/Equal_Sides_Of_An_Array.ts
export const findEvenIndex = (arr: number[]): number => {
  let left = 0,
    right = arr.reduce((a, b) => a + b, 0);
  for (let i = 0; left <= right; ++i) {
    right -= arr[i];
    if (left === right) {
      return i;
    }
    left += arr[i];
  }
  return -1;
};

File: rust/src/6kyu/Equal_Sides_Of_An_Array.rs
fn find_even_index(arr: &[i32]) -> Option<usize> {
    let (mut r, mut l) = (arr.iter().sum::<i32>(), 0);
    for (i, n) in arr.iter().enumerate() {
        r -= n;
        if r == l { return Some(i) }
        l += n;
    }
    None
}